libarchive/archive_read_open_file.c

```diff
@@ -150,7 +150,9 @@ file_skip(struct archive *a, void *client_data, int64_t request)
 			skip = max_skip;
 	}
 
-#if HAVE_FSEEKO
+#ifdef __ANDROID__
+	if (lseek(fileno(mine->f), skip, SEEK_CUR) < 0)
```

Comment:
Relying on fileno(mine->f) is a bad idea in general, since FILE on some systems is an abstract interface that can be used to wrap things that are not, in fact, files.

It's okay in this case since the code will only run on Android -- I presume Android's FILE is not so complicated.